### ISSUE TITLE
JupyterLab 4 / nbgrader 0.9 support

### ISSUE THREAD

**Issue by rubdos** [lxylxy123456/ngshare#163]
In September, the [0.9-series of nbgrader was released](https://github.com/jupyter/nbgrader/releases/tag/v0.9.0) (meanwhile 0.9.1 came out too). This version is compatible with JupyterLab 4.

As far as I know, current ngshare works on JupyterLab 3 only. It would be cool if we could start rolling out JL4/nbg0.9 starting next academic year! I shamefully don't have any idea about the complexity of the migration that's necessary for this, though.

**Comment by rmoreas** [comment]
Seems that nbshare project is dead?

**Comment by lahwaacz** [comment]
> As far as I know, current ngshare works on JupyterLab 3 only.

@rubdos Did you mean JupyterHub rather than JupyterLab here?

**Comment by rubdos** [comment]
> @rubdos Did you mean JupyterHub rather than JupyterLab here?

Yes, most probably. The versioning is rather confusing, especially if I only look a few times a year at it...

**Comment by lahwaacz** [comment]
So ngshare [depends on jupyterhub](https://github.com/LibreTexts/ngshare/blob/master/setup.py#L45) only and does not care about the jupyterlab version (there is no interaction with its UI directly). I tested ngshare with jupyterhub versions 4 and 5 and did not find any compatibility problems.

While the `setup.py` does not specify a dependency on nbgrader, it works with the latest nbgrader version 0.9.3 (and the current `main` branch actually). The only issues I had while setting up ngshare were related to the complexity of nbgrader configuration and bugs in nbgrader itself...
